trunner/target/armv7a9.py

```diff
@@ -71,8 +73,14 @@ def __call__(self):
             except subprocess.TimeoutExpired as e:
                 raise FlashError(msg=str(e), output=e.stdout.decode("ascii") if e.stdout else None) from e
 
-        self.enter_bootloader()
-        self.wait_prompt()
+        try:
+            self.dut.expect_exact("Waiting for input", timeout=0.5)
+            self.enter_bootloader()
+        except pexpect.TIMEOUT as e:
+            if "\x1b[31m" in self.dut.before:
```

Comment:
Why you are expecting red color? There may be some other error - we want to catch it. I'd let the magic error happen and leave a comment why.